From 272713d611c55effeaf1d2b45fbb2c69d70e4c74 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org>
Date: Sat, 23 Dec 2017 21:27:20 +0000
Subject: Fix build with --enable-gdbm-export

* THANKS: Add Jakub Bogusz
* export/export.c: Define GDBM_EXPORT_18 prior to
including gdbmexp.c
* src/gdbmexp.c [GDBM_EXPORT_18]: Define GDBM_SET_ERRNO and
GDBM_ASSERT_CONSISTENCY.
---
diff --git a/THANKS b/THANKS
index 70cb40f..feb1f87 100644
--- a/THANKS
+++ b/THANKS
@@ -5,6 +5,7 @@ suggesting various improvements or submitting actual code. Here is a list
 of these people. Help us keep it complete and exempt of errors.
 
 Bill Jones              <rj7252@att.com>
+Jakub Bogusz            <qboosh@pld-linux.org>
 Matthew Burgess		<matthew@linuxfromscratch.org>
 Tanaka Akira 		<akr@fsij.org>
-Thomas Klausner		<tk@giga.or.at>
\ No newline at end of file
+Thomas Klausner		<tk@giga.or.at>
diff --git a/export/export.c b/export/export.c
index 39e05d1..ede4858 100644
--- a/export/export.c
+++ b/export/export.c
@@ -24,6 +24,7 @@
 #include <gdbm.h>
 
 /* Pull in gdbm_export() */
+#define GDBM_EXPORT_18
 #include "gdbmexp.c"
 
 void
diff --git a/src/gdbmexp.c b/src/gdbmexp.c
index 457f638..dc46c50 100644
--- a/src/gdbmexp.c
+++ b/src/gdbmexp.c
@@ -21,8 +21,13 @@
 # include "autoconf.h"
 # include <arpa/inet.h>
 
+#ifdef GDBM_EXPORT_18
+# define GDBM_SET_ERRNO(dbf, ec, fatal) gdbm_errno = ec
+# define GDBM_ASSERT_CONSISTENCY(dbf, val)
+#else
 # include "gdbmdefs.h"
 # include "gdbm.h"
+#endif
 
 int
 gdbm_export_to_file (GDBM_FILE dbf, FILE *fp)
--
cgit v0.9.0.3
